### DISCUSSION TITLE
Use `http2` directive instead of `listen ... http2`

### DISCUSSION THREAD

**Discussion by chubei** [nginx/kubernetes-ingress#4117] [Ideas]
**Is your feature request related to a problem? Please describe.**

`nginx-ingress` currently doesn't support enabling HTTP2 unless using TLS.

**Describe the solution you'd like**

In nginx 1.25.1, HTTP/1.1 and HTTP/2 on same TCP stream is supported, with the new `http2` directive. We can use it in the latest `nginx-ingress`.

**Describe alternatives you've considered**

None.

**Additional context**

None.

**Comment by brianehlert** [comment]
There are a number of changes that are happening around the http2 directives, if you have not noticed.
Right now we are evaluating how to align with http2 directive deprecation as well as new directives and patterns.
I think this is just one part of that larger amount of effort.

**Reply by brianehlert** [reply]
Added work item here: https://github.com/nginxinc/kubernetes-ingress/issues/4237
